tensor_genn/algorithms/relu_ann.py

```diff
@@ -0,0 +1,129 @@
+        
+        for layer in tf_model.layers:
+            if not isinstance(layer,(tf.keras.layers.Dense,tf.keras.layers.Flatten)):
+                raise NotImplementedError('Only Dense layers are supported')
+        if len(tf_model.get_weights()) >= len(tf_model.layers):
+            raise NotImplementedError('Tensorflow model should be trained without biases')
+
+        # create custom classes
+        if_model = genn_model.create_custom_neuron_class(
+            "if_model",
+            param_names=["Vres","Vthr","Cm"],
+            var_name_types=[("Vmem","scalar"),("SpikeNumber","unsigned int")],
+            sim_code="""
+            $(Vmem) += $(Isyn)*(DT / $(Cm));
+            """,
+            reset_code="""
+            $(Vmem) = $(Vres); 
+            $(SpikeNumber) += 1;
+            """,
+            threshold_condition_code="$(Vmem) >= $(Vthr)"
+        )
+
+        cs_model = genn_model.create_custom_current_source_class(
+            "cs_model",
+            var_name_types=[("magnitude","scalar")],
+            injection_code="""
+            $(injectCurrent, $(magnitude));
+            """
+        )
+
+        # Fetch tf_model details
+        n_layers = len(tf_model.layers)
+        tf_weights = tf_model.get_weights()
+
+        # Params and init
+        if_params = {
+            "Vres":self.Vres,
+            "Vthr":self.Vthr,
+            "Cm":self.Cm
+        }
+        if_init = {
+            "Vmem":genn_model.init_var("Uniform", {"min": self.Vres, "max": self.Vthr}),
+            "SpikeNumber":0
+        }
+        
+        cs_init = {"magnitude":10.0}
+
+        # Define model and populations
+        self.g_model = genn_model.GeNNModel("float","g_model")
+        self.neuron_pops = []
+
+        for i in range(1,n_layers): # 1,2 - synapses
+            if i == 1:
+                # Presynaptic neuron
+                self.neuron_pops.append(self.g_model.add_neuron_population(
+                    "if"+str(i-1),tf_weights[i-1].shape[0],if_model,if_params,if_init)
+                )
+
+            # Postsynaptic neuron
+            self.neuron_pops.append(self.g_model.add_neuron_population(
+                "if"+str(i),tf_weights[i-1].shape[1],if_model,if_params,if_init)
+            )
+
+            # Synapse
+            self.g_model.add_synapse_population(
+                "syn"+str(i-1)+str(i),"DENSE_INDIVIDUALG",genn_wrapper.NO_DELAY,
+                self.neuron_pops[i-1], self.neuron_pops[i],
+                "StaticPulse",{},{'g':tf_weights[i-1].reshape(-1)},{},{},
+                "DeltaCurr",{},{}
+            )
+        
+        self.current_source = self.g_model.add_current_source("cs",cs_model,"if0",{},cs_init)
+
+        self.g_model.dT = self.timestep
+        self.g_model.build()
+        self.g_model.load()
+
+        return self.g_model
+
+    def evaluate(self, X, y=None):
+        n_examples = len(X)
+        X = X.reshape(n_examples,-1)
```

Comment:
Sorry for being daft, but what is this doing?

Reply:
It converts the input X to the desired shape i.e. (num_examples, num_features). I've done this so that the method doesn't fail when images are passed as, say (60000, 28,28)